epoxy-adapter/src/main/java/com/airbnb/epoxy/ChangeAnimationDelegator.java

```diff
@@ -0,0 +1,168 @@
+import android.support.v7.widget.RecyclerView.ViewHolder;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * ItemAnimator that wraps any other ItemAnimator and delegates change animations to
+ * {@link EpoxyModel}
+ */
+public final class ChangeAnimationDelegator extends ItemAnimatorDecorator
+    implements AnimatorListener {
+  /* use a BiMap so we can use the same listener for every animator and quickly find the holder for
+     an animation. We do this to be able to remove only OUR listener from an animator and not those
+     that the user might have set */
+  @NonNull private final BiHashMap<ViewHolder, EpoxyItemAnimation> mChangeAnimations =
+      new BiHashMap<>();
+
+  public ChangeAnimationDelegator(@NonNull ItemAnimator decoratedAnimator) {
+    super(decoratedAnimator);
+  }
+
+  public long getChangeDuration() {
+    long maxDuration = super.getChangeDuration();
+
+    for (EpoxyItemAnimation anim : mChangeAnimations.values()) {
+      maxDuration = Math.max(maxDuration, anim.getDuration());
+    }
+
+    return maxDuration;
+  }
+
+  public void onAnimationEnd(EpoxyItemAnimation anim) {
+    dispatchAnimationFinished(mChangeAnimations.inverse().get(anim));
+    mChangeAnimations.inverse().remove(anim);
+
+    if (mChangeAnimations.isEmpty()) {
+      dispatchChangeAnimationsFinished();
+    }
+  }
+
+  @Override
+  public void onAnimationCancel(EpoxyItemAnimation anim) {
+    mChangeAnimations.inverse().remove(anim);
+  }
+
+  public void onAnimationStart(EpoxyItemAnimation anim) {
+    dispatchAnimationStarted(mChangeAnimations.inverse().get(anim));
+  }
+
+  @SuppressWarnings("unchecked")
+  public boolean animateChange(
+      @NonNull final ViewHolder oldHolder, @NonNull final ViewHolder newHolder,
+      @NonNull ItemHolderInfo preLayoutInfo, @NonNull ItemHolderInfo postLayoutInfo) {
+
+    if (oldHolder == newHolder && oldHolder instanceof EpoxyViewHolder
+        && canHolderAnimateChanges((EpoxyViewHolder) oldHolder)) {
+
+      final EpoxyViewHolder holder = ((EpoxyViewHolder) oldHolder);
+      final EpoxyModel newModel = holder.getModel();
+      final Object viewToBind = holder.objectToBind();
+      final EpoxyModel oldModel = DiffPayload.getModelFromPayload();
+
+      final EpoxyItemAnimation oldAnim = mChangeAnimations.get(oldHolder);
+
+      if (oldAnim == null || !oldAnim.isRunning()) {
+        EpoxyItemAnimation newAnim = newModel.bindChangesAnimated(viewToBind, oldModel);
+        newAnim.addListener(this);
+        mChangeAnimations.put(holder, newAnim);
+      } else {
+        EpoxyItemAnimation continuedAnim =
+            newModel.bindChangesAnimated(viewToBind, oldModel, oldAnim);
+
+        if (continuedAnim != null) {
+          //cancel old anim if user decided to create a new animator instead of reusing the old one
+          if (continuedAnim != oldAnim) {
+            oldAnim.cancel();
+            continuedAnim.start();
+          }
+
+          continuedAnim.addListener(this);
```

Comment:
Good spot. onStart would call `dispatchAnimationStarted`, but we already did that for the interrupted animation... Might cause a bug. I'll have to consult the example by the RV creators.